perfkitbenchmarker/benchmarks/cassandra_stress_benchmark.py

```diff
@@ -174,100 +304,151 @@ def Prepare(benchmark_spec):
+                    cassandra_stress_profile_ops,
+                    pop_per_vm,
+                    cassandra_stress_pop_dist,
+                    cassandra_stress_pop_params):
   """Run Cassandra-stress test on loader node.
 
   Args:
     vm: The target vm.
     loader_index: The index of target vm in loader vms.
-    keys_per_vm: The number of keys per loader vm need to insert.
+    ops_per_vm: The number of operations each loader vm requests.
     data_node_ips: List of IP addresses for all data nodes.
+    cassandra_stress_command: The cassandra-stress command to use.
+    cassandra_stress_profile_ops: The ops to use with user mode.
+    pop_per_vm: Population per loader vm.
+    cassandra_stress_pop_dist: The population distribution.
+    cassandra_stress_pop_params: String representing additional population
+        parameters.
   """
-  cassandra_stress_command = FLAGS.cassandra_stress_command
-
   if cassandra_stress_command == USER_COMMAND:
     cassandra_stress_command += ' profile={profile} ops\({ops}\)'.format(
         profile=TEMP_PROFILE_PATH,
-        ops=FLAGS.cassandra_stress_ops)
+        ops=cassandra_stress_profile_ops)
     schema_option = ''
   else:
+    if cassandra_stress_command == MIXED_COMMAND:
+      cassandra_stress_command += ' ratio\({ratio}\)'.format(
+          ratio=FLAGS.cassandra_stress_mixed_ratio)
     # TODO: Support more complex replication strategy.
     schema_option = '-schema replication\(factor={replication_factor}\)'.format(
         replication_factor=FLAGS.cassandra_stress_replication_factor)
+
+  pop_params = '%s..%s' % (loader_index * pop_per_vm + 1,
+                           (loader_index + 1) * pop_per_vm)
+  if cassandra_stress_pop_params:
+    pop_params = '%s,%s' % (pop_params, cassandra_stress_pop_params)
+  if cassandra_stress_pop_dist:
+    pop_dist = 'dist=%s\(%s\)' % (
+        cassandra_stress_pop_dist, pop_params)
+  else:
+    pop_dist = 'seq=%s' % pop_params
   vm.RobustRemoteCommand(
       '{cassandra} {command} cl={consistency_level} n={num_keys} '
-      '-node {nodes} {schema} -pop seq={start_index}..{end_index} '
+      '-node {nodes} {schema} -pop {pop_dist} '
       '-log file={result_file} -rate threads={threads} '
       '-errors retries={retries}'.format(
           cassandra=CASSANDRA_STRESS,
           command=cassandra_stress_command,
           consistency_level=FLAGS.cassandra_stress_consistency_level,
-          num_keys=keys_per_vm,
+          num_keys=ops_per_vm,
           nodes=','.join(data_node_ips),
           schema=schema_option,
-          start_index=loader_index * keys_per_vm + 1,
-          end_index=(loader_index + 1) * keys_per_vm,
+          pop_dist=pop_dist,
           result_file=_ResultFilePath(vm),
           retries=FLAGS.cassandra_stress_retries,
           threads=FLAGS.num_cassandra_stress_threads))
 
 
-def RunCassandraStress(benchmark_spec):
-  """Start Cassandra test.
+def RunCassandraStressTest(benchmark_spec,
```

Comment:
Right. So can you remove the benchmark spec as an argument, and instead pass in the individual components needed? Or is that too many arguments?